examples/error-subtyping/error_subtyping.bal

```diff
@@ -1,21 +1,46 @@
 import ballerina/io;
 
-// `distinct` creates a new subtype.
-type XErr distinct error;
-type YErr distinct error;
+type InvalidIntDetail record {|
+    int value;
+|};
 
-type Err XErr|YErr;
+type InvalidI32Detail record {|
+    int:Signed32 value;
+|};
 
-// The name of the distinct type can be used with the `error` constructor to create an error value 
-// of that type. `err` holds an `error` value of type `XErr`.
-Err err = error XErr("Whoops!");
+type InvalidIntError error<InvalidIntDetail>;
 
-function desc(Err err) returns string {
-    // The `is` operator can be used to distinguish distinct subtypes.
-    return err is XErr ? "X" : "Y";
+type InvalidI32Error error<InvalidI32Detail>;
 
+type DistinctIntError distinct error<InvalidIntDetail>;
+
+type AnotherDistinctIntError distinct error<InvalidIntDetail>;
```

Comment:
IMO comments for these definitions (subtypes of what) can be helpful.

Reply:
Fixed